lib/web_ui/lib/src/engine/text_editing/input_type.dart

```diff
@@ -92,6 +92,9 @@ class NoTextInputType extends EngineInputType {
 
   @override
   String get inputmodeAttribute => 'none';
+
+  @override
+  DomHTMLElement createDomElement() => createDomHTMLTextAreaElement();
```

Comment:
So `TextInputType.none` is always treated as multiline? I don't think that's correct. cc @Renzo-Olivares for ideas on how to distinguish between single-line and multi-line modes when using `TextInputType.none`.

Reply:
Agreed here.  What's the reasoning behind setting `none` as an input type for [custom text input controls](https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/services/text_input.dart#L2229) in the framework?  Could we just make the determination at the framework level (via `maxLines > 1` or something similar) and just send an explicit input type to the engine?

Reply:
I agreed to solve via framework level.
However, I think, the main reason for sending `none` as an input type is to disable the soft keyboard display. 
So, we may add a property (such as disableSoftKeyboard) to the class `InputConfiguration`. Then we will add code to all platforms to handle this property to disable soft keyboard display.

Reply:
Agreeing with @Satsrag, I think we need to continue setting the `inputMode` of the web dom element (input/textarea) to `none` to block the virtual keyboard when a custom text input control is provided. This means we can't conditionally send an explicit input type based on `maxLines` (unless we add more input types).

Adding a flag to `TextInputConfiguration` as suggested by @Satsrag sounds reasonable, but `disableSoftKeyboard` sounds like it would accomplish the same thing as passing [`TextInputType.none`](https://api.flutter.dev/flutter/services/TextInputType/none-constant.html). Maybe what we need is a `forceMultiline/isMultiline` parameter which sets its value based on the `maxLines` field. Then the web engine can use this to decide whether to use the  `input/textarea`.

Reply:
Yes, I agree with @Renzo-Olivares.